var documenterSearchIndex = {"docs":
[{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"Getting Started This chapter covers everything you need to know to start using MP-BioPath to analyze your files. This includes installing MP-BioPath and running a few example commands.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"Installation To simplify the installation process, a Docker container has been generated. All you need to do is pull the latest version of the container.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"If you're not familiar with Docker, it provides an environment with all the dependencies pre-installed to run the tool.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"Docker Docker is a program that runs on your computer. You'll need to install it on your OS X, Windows, or Linux system.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"Instructions for installing Docker can be found here: Docker Installation Guide","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"Docker Image To obtain the latest Docker container, use the following command:","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"bash Copy code: docker pull oicr/mp-biopath:1.0.x","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"After running this command, check that you have the container by typing:","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"bash Copy code: docker images","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"You should see the image \"oicr/mp-biopath\" in your list of images.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"Running the Container To run a command line from within the container, use:","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"bash Copy code: make run-bash","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"This command places you inside the container. Note that the filesystem inside the container is separate from the host filesystem. The \"-v\" flag maps a directory outside the container to a folder inside the container. In this example, it maps the current directory to a \"data\" directory inside the container. Changes to files made within this directory will not be lost when exiting the container.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"To run commands inside the container from the host, use:","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"bash Copy code: docker run mpbiopath julia test/run.jl","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"This command runs the tests for the tool. For running custom commands, refer to the \"Running MP-BioPath\" section.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"Custom Setup To run the tool on your own data, you need to create evidence files in the specified format. You also need to obtain the Reactome pathways in pairwise interaction format from Reactome Pathway Analysis.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"After obtaining the required files, create a custom configuration file. MP-BioPath will use this configuration file to run and produce results in the specified folder.","category":"page"},{"location":"getting-started/","page":"Getting Started","title":"Getting Started","text":"Refer to specific sections of this document for details on developing files in the correct format.","category":"page"},{"location":"results/","page":"Results","title":"Results","text":"Results","category":"page"},{"location":"results/","page":"Results","title":"Results","text":"The results will be output in the same format as the evidence files, where there will be a row for each of the nodes in the pathway and the columns will match the samples.","category":"page"},{"location":"results/","page":"Results","title":"Results","text":"These files will be located in a subfolder for each of the pathways specified in the run.","category":"page"},{"location":"results/","page":"Results","title":"Results","text":"From these results, there are tools for generating heatmaps and performing further analysis. We intend to provide a more robust set of tools for analyzing the results generated by this program.","category":"page"},{"location":"expression/","page":"Expression","title":"Expression","text":"# Expression Module\n\nThis module provides functions for reading expression data from files and extracting data for a specific tissue.\n\n## Usage\n\nTo use the `Expression` module, first import it:\n","category":"page"},{"location":"expression/","page":"Expression","title":"Expression","text":"julia using Expression","category":"page"},{"location":"expression/","page":"Expression","title":"Expression","text":"\n### `getExpression(filename)`\n\nReads expression data from a file and returns a DataFrame.\n\n- `filename::AbstractString`: The path to the file containing expression data.\n\nExample:","category":"page"},{"location":"expression/","page":"Expression","title":"Expression","text":"julia expressiondata = getExpression(\"path/to/expressiondata.tsv\")","category":"page"},{"location":"expression/","page":"Expression","title":"Expression","text":"\n### `getTissue(df, tissue)`\n\nExtracts expression data for a specific tissue from a DataFrame.\n\n- `df::DataFrame`: The DataFrame containing expression data.\n- `tissue::AbstractString`: The column name corresponding to the tissue of interest.\n\nExample:","category":"page"},{"location":"expression/","page":"Expression","title":"Expression","text":"julia tissuedata = getTissue(expressiondata, \"Tissue1\")","category":"page"},{"location":"expression/","page":"Expression","title":"Expression","text":"\nReplace `\"path/to/expression_data.tsv\"` with the actual path to your expression data file.\n\nFor more information on working with DataFrames, refer to the [DataFrames.jl documentation](https://dataframes.juliadata.org/stable/).","category":"page"},{"location":"introduction/","page":"Introduction","title":"Introduction","text":"MP-BioPath (Mathmatical Programming Biological Pathways)","category":"page"},{"location":"introduction/","page":"Introduction","title":"Introduction","text":"Developed by: Adam Wright","category":"page"},{"location":"introduction/","page":"Introduction","title":"Introduction","text":"Contributors: Thomas Barbazuk","category":"page"},{"location":"introduction/","page":"Introduction","title":"Introduction","text":"Version v1.0 (Research Edition) - December 28th 2017","category":"page"},{"location":"introduction/","page":"Introduction","title":"Introduction","text":"Date: April 12 2024","category":"page"},{"location":"introduction/","page":"Introduction","title":"Introduction","text":"Documentation: April 12 2024","category":"page"},{"location":"introduction/","page":"Introduction","title":"Introduction","text":"MP-BioPath is a biological pathway tool used to infer the activity of individual pathways. The tool implements a novel approach, using Mathematical Programming, to infer pathway activity. It can read a combination of copy number, mutation, expression level, and proteomic data to make these inferences. The tool uses evidence of biological perturbations to infer the activity of the remaining nodes in the specified pathway.","category":"page"},{"location":"","page":"Home","title":"Home","text":"Hi again","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Pathways In this chapter we will be going over the pathway file format for MP-BioPath.","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"OICR Pathway Tab format This pairwise intearciton tab delimited format reduces the information on the interactions into a computable format.","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Simple Example:","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"5619379    6810158RLE    1    0 5617644    6810158RLE    1    0 The file consists of four types of information: entity and interactions. For each interaction there is a parent and a child node and for each pair there is relationship information that specified whether the node is in an \"AND\" or an \"OR\" relationship. And if the interaction is a postitive or negative interaction. In this example we have two entities having positibe AND relationship going into a reaction node.","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Entities There is a controlled vocabulary of entity types that can be used as part of a MP-BioPath pathway. These entity types are: gene, rna, dna, protein,If a protein does not have a a preceding RNA entity, one will be created at runtime. And if an RNA entity does not have a preceding genome entity, one will be created at runtime.","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"A pathway can compose of the following entity types:","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Simple Entities Gene This is actual gene itself that is used to create RNA","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"RNA This is the RNA that is used to create the protein","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Protein protein nodes often interact with other proteins through reactions to effect the functioning of a specific pathway.","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Abstract Entities Set This entity is used in a pathway when one of many entities could be used to fulfil the same (or similar) function. A set (or Family) is comosed of many \"members\" or have two sources of the same entity.","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Complex A complex is an entity that is composed of other entites. These entities are known as components of the complex.","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Interaction Formats The first column in contains the parent node. And the second column contains the child node of the interaction. The following are use to specify the information on how the parent child pair interact:","category":"page"},{"location":"pathways/","page":"Pathways","title":"Pathways","text":"Activating Interaction 1 (Third column) Inhibition Interaction -1 (Third column) AND relationship 0(Fourth column) Inhibition Interaction 1(Fourth column)","category":"page"}]
}
